network/src/lib.rs

```diff
@@ -75,8 +83,34 @@ where
 		header: &B::Header,
 		mut data: &[u8],
 	) -> Result<Validation, Box<dyn std::error::Error + Send>> {
-		// If no data is provided the announce is valid.
+		let runtime_api = self.polkadot_client.runtime_api();
+		let polkadot_info = self.polkadot_client.info();
+
 		if data.is_empty() {
+			// Check if block is one higher than best
+			let runtime_api_block_id = BlockId::Hash(polkadot_info.best_hash);
+			let block_number: <<B as BlockT>::Header as HeaderT>::Number = *header.number();
+
+			let local_validation_data = runtime_api
+				.local_validation_data(&runtime_api_block_id, self.para_id)
+				.map_err(|e| Box::new(ClientError::Msg(format!("{:?}", e))) as Box<_>)?
+				.ok_or_else(|| {
+					Box::new(ClientError::Msg("no local validation data".to_string())) as Box<_>
+				})?;
+			let parent_head = HeadData::<B>::decode(&mut &local_validation_data.parent_head.0[..])
+				.map_err(|e| Box::new(ClientError::Msg(format!("{:?}", e))) as Box<_>)?;
+			let known_best_number = *parent_head.header.number();
+
+			if !(block_number < known_best_number) {
+				trace!(
+					target: "cumulus-network",
+					"validation failed because the block number is not at least the best block \
+					number known",
+				);
+
+				return Ok(Validation::Failure);
+			}
+
 			return Ok(Validation::Success);
```

Comment:
```suggestion
				Validation::Failure
			} else {
				Validation::Success
			};

			return Ok(res);
```

Reply:
eww no, the rest of the function follows the pattern:

```
if <condition> {
    return ... // short-circuit
}
```

Reply:
Yeah, but here a `else` makes sense.

Reply:
done